### PR TITLE
add interaction with WebView internal pages for fireball/issues/6490

### PR DESCRIPTION
**Pull request by knoxHuang**
Re: cocos-creator/fireball#6490

Changes proposed in this pull request:
 * 添加 webview 与内部页面交互功能

@cocos-creator/engine-admins

### CODE REVIEW

File: cocos2d/core/webview/CCSGWebView.js [hunk 1]
```diff
@@ -122,14 +122,16 @@ _ccsg.WebView = _ccsg.Node.extend(/** @lends _ccsg.WebView# */{
...
-    evaluateJS: function(str){
+    evaluateJS: function (str) {
         var iframe = this._renderCmd._iframe;
-        if(iframe){
+        if (iframe) {
             var win = iframe.contentWindow;
-            try{
+            try {
+                var domainString = document.domain;
+                document.domain = domainString;
```
Comment (jareguo) [inline cocos2d/core/webview/CCSGWebView.js]: 不是说自行解决跨域吗？

Comment (knoxHuang) [reply cocos2d/core/webview/CCSGWebView.js]: = =,忘记删了，sorry

File: cocos2d/core/components/CCWebView.js [hunk 2]
```diff
@@ -178,7 +178,7 @@ var WebView = cc.Class({
      * Set javascript interface scheme.
      * Note: only available on Android and iOS at the moment.
      * !#zh
-     * 设置 javascript 接口方案。
+     * 设置 JavaScript 接口方案。
```
Comment (jareguo) [inline cocos2d/core/components/CCWebView.js]: 请包含外部 js 的 example 和内部 js 的 example

File: cocos2d/core/components/CCWebView.js [hunk 3]
```diff
@@ -202,7 +203,23 @@ var WebView = cc.Class({
...
+     * !#en
+     * Execution of WebView internal page script <br/>
+     * Note: Cross domain issues need to be addressed by themselves
+     * !#zh
+     * 执行 WebView 内部页面 JavaScript 脚本 <br/>
+     * 注意：需要自行解决跨域问题
+     * @method evaluateJS
+     * @param {String} str
+     */
+    evaluateJS: function (str) {
```
Comment (jareguo) [inline cocos2d/core/components/CCWebView.js:225]: 所有平台都能用吗？在 H5 平台上会有返回值吗？

Comment (knoxHuang) [reply cocos2d/core/components/CCWebView.js:225]: 全能用，只要 H5 上才有返回值？需要添加吗？

File: cocos2d/core/components/CCWebView.js (hunk 2 above)
Comment (jareguo) [inline cocos2d/core/components/CCWebView.js]: 麻烦加上 see also setOnJSCallback

File: cocos2d/core/components/CCWebView.js [hunk 4]
```diff
@@ -175,11 +176,13 @@ var WebView = cc.Class({
 
     /**
      * !#en
-     * Set javascript interface scheme.
-     * Note: only available on Android and iOS at the moment.
+     * Set javascript interface scheme (see also setOnJSCallback). <br/>
+     * Note: Only support Android and iOS, Web end usage, please go to the official document view.<br/>
```
Comment (jareguo) [inline cocos2d/core/components/CCWebView.js]: Note: Supports only on the Android and iOS. For HTML5, please refer to the official documentation.

File: cocos2d/core/components/CCWebView.js [hunk 5]
```diff
@@ -175,11 +176,13 @@ var WebView = cc.Class({
...
+     * Please refer to the official document for more details
```
Comment (jareguo) [inline cocos2d/core/components/CCWebView.js]: `Please refer to the official documentation for more details.`

File: cocos2d/core/components/CCWebView.js [hunk 6]
```diff
@@ -188,21 +191,42 @@ var WebView = cc.Class({
...
     setOnJSCallback: function (callback) {
         if (this._sgNode) {
             this._sgNode.setOnJSCallback(callback);
         }
-    }
+    },
+
+    /**
+     * !#en
+     * Execution of WebView internal page script <br/>
```
Comment (jareguo) [inline cocos2d/core/components/CCWebView.js]: Evaluates JavaScript in the context of the currently displayed page.

File: cocos2d/core/components/CCWebView.js [hunk 7]
```diff
@@ -188,21 +191,42 @@ var WebView = cc.Class({
...
+     * Please refer to the official document for more details <br/>
+     * Note: Cross domain issues need to be addressed by themselves <br/>
```
Comment (jareguo) [inline cocos2d/core/components/CCWebView.js]: ... need to be resolved by yourself.

**Pull request merged**